internal/controller/pkg/revision/hook.go

```diff
@@ -158,6 +165,9 @@ func (h *ProviderHooks) Post(ctx context.Context, pkg runtime.Object, pr v1.Pack
 	if err := initializer.NewTLSCertificateGenerator(h.namespace, initializer.RootCACertSecretName, *pr.GetTLSServerSecretName(), *pr.GetTLSClientSecretName(), pkgProvider.Name, initializer.TLSCertificateGeneratorWithOwner(owner)).Run(ctx, h.client); err != nil {
 		return errors.Wrapf(err, "cannot generate TLS certificates for %s", pkgProvider.Name)
 	}
+	if err := h.client.Apply(ctx, d); err != nil {
+		return errors.Wrap(err, errApplyProviderDeployment)
+	}
```

Comment:
I guess this was moved here because you want to be sure providers are created after the certificates are generated, right?

Reply:
yes, that's the reason